### DISCUSSION TITLE
How to get inspector instance inside on change graph

### DISCUSSION THREAD

**Discussion by pandika-csgi** [clientIO/joint#2221] [Q&A] [answered]
### Introduction

Hi,

I want to update my custom input inside inspector, there is a method called `updateCustomInput(inspector: ui.Inspector, $customField: any, options: any, path: string)` inside `inspector-service.ts`

Update triggered by toggle inside inspector called `toggleCustomInput`

I've success listened change for the `toggleCustomInput` toggle via `this.graph.on('change', (cell: dia.Link, value, options) => { ... } ` but I need inspector instance when call `updateCustomInput`, I got the inspector id inside `options.inspector` 

How to get an inspector instance inside `graph.on('change')`?
I'm using JointJS v3.6.5 

Thank you

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS, JointJS+

**Comment by kumilingus** [comment, marked as answer]
You need to store the reference to the inspector instance when you create it.

```js
let inspector;

// ...

inspector =  new ui.Inspector();

// ...

graph.on('change', (cell, opt) => {
   if (inspector && inspector.cid === opt.inspector) {
      // the change was made by this inspector
   }
});
```
